test/Test/Data/Map.purs

```diff
@@ -86,16 +75,7 @@ instance showInstruction :: (Show k, Show v) => Show (Instruction k v) where
   show (Delete k) = "Delete (" <> show k <> ")"
 
 instance arbInstruction :: (Arbitrary k, Arbitrary v) => Arbitrary (Instruction k v) where
-  arbitrary = do
-    b <- arbitrary
-    case b of
-      true -> do
-        k <- arbitrary
-        v <- arbitrary
-        pure (Insert k v)
-      false -> do
-        k <- arbitrary
-        pure (Delete k)
+  arbitrary = oneOf (Insert <$> arbitrary <*> arbitrary) [(Delete <$> arbitrary)]
```

Comment:
Oops, missed this before - could you please remove the redundant parens around `Delete`?